### DISCUSSION TITLE
How to color values of "0" instead of treating as none

### DISCUSSION THREAD

**Discussion by hfschwing** [Richardsl/heatmap-calendar-obsidian#133] [Q&A]
I am plotting data that has values of 0 to 8 where 0 is different than no value(null). I do have some entries that have no data too but they are just no entry and not a valid value of 0. The plugin seems to treat zero as no value. So a zero value plots with the background color just like a no value and not the first color in my color list. Is there a way to treat zero and no value differently and plot a color for 0?

**Comment by Asi0Flammeus** [comment]
Don't know if it can still be relevant to someone but I had the same issue and I found a workaround by simply adding 1 to the monitored value when rendering the heatmap. 
So it allows me to keep the same initially desired scaled (from 0 to 10) but the heatmap will displayed it as a shifted scale (from 1 to 11). 

Note that is also work with checkbox property types (1 = false / not checked, 2 = true / checked).  

Here's an example:

```dataviewjs
dv.span("** Social Media **")

const hue1 = 13  //red
const hue2 = 132 //green

const calendarData = {
    year: 2025, // optional, remove this line to autoswitch year 
    intensityScaleStart: 1,
    intensityScaleEnd: 10,
    colors: {
        red2green: [
            `hsl(${hue2}, 70%, 75%)`,      // 0 - lightest green
            `hsl(${hue1}, 70%, 45%)`,      // 1 - darker green
            `hsl(${hue1}, 100%, 77%)`,     // 2 - lightest red
            `hsl(${hue1}, 100%, 60%)`,     // 3 - 
            `hsl(${hue1}, 100%, 50%)`,     // 4 - 
            `hsl(${hue1}, 100%, 37%)`,     // 5 - 
            `hsl(${hue1}, 89%, 24%)`,      // 6 - 
            `hsl(${hue1}, 59%, 24%)`,      // 7 - 
            `hsl(${hue1}, 49%, 24%)`,      // 8+ - darkest red
        ],
    },
    entries: []
}

for(let page of dv.pages('"02 Roles"').where(p=>p.no_CSM+1)){ 
    calendarData.entries.push({
        date: page.file.name, 
        intensity: page.no_CSM + 1,
        content: await dv.span(`[](${page.file.name})`), //for hover preview
    })
}

renderHeatmapCalendar(this.container, calendarData)
```
![image](https://github.com/user-attachments/assets/f62f6ee1-c0a5-4f9a-b5d2-959640a541f3)
